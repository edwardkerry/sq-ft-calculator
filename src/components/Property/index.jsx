import React from 'react';
import { string, number } from 'prop-types';

const Property = ({address, price, floor_area}) => (
  <li className='property'>
    <p className='address'>Address: {address}</p>
    <p className='price'>Value: Â£{price}</p>
    <p className='floor_area'>Floor area(sq ft): {floor_area}</p>
  </li>
);

Property.propTypes = {
  address: string.isRequired,
  price: number. isRequired,
  floor_area: number.isRequired
};

export default Property;
