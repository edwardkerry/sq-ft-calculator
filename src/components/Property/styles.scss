li {
  margin: 5px;
  padding: 20px;
  border: 2px solid rgb(53, 58, 57);
  background-color: rgb(169, 170, 177);
}

.active {
  border: 2px solid rgb(65, 221, 73);
  background-color: rgb(113, 228, 113);
}
