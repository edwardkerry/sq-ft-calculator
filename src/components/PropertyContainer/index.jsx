import React from 'react';
import { arrayOf, shape, string, number } from 'prop-types';
import Property from '../Property';

const PropertyContainer = ({properties}) => (
  <ul className='properties'>
    {properties.map((property, index) => {
        return (<Property
          address={property.address}
          price={property.price}
          floor_area={property.floor_area}
          key={index}
        />)
      })}
  </ul>
);

PropertyContainer.propTypes = {
  properties: arrayOf(
    shape({
      address: string.isRequired,
      price: number.isRequired,
      floor_area: number.isRequired
    }).isRequired
  ).isRequired
};

export default PropertyContainer;
