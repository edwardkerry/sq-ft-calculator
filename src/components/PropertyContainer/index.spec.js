import React from 'react';
import renderer from 'react-test-renderer';
import propertyList from'../../testSetup/propertyFixtures'
import PropertyContainer from './';
import Property from '../Property';

describe('Property Container', () => {
  it('renders', () => {
    const component = renderer.create(< PropertyContainer properties = { propertyList } />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  });

  it('displays a Property component for each passed property', () => {
    const wrapper = shallow(<PropertyContainer properties={propertyList} />)
    expect(wrapper.find('.properties').children().length).toEqual(propertyList.length)
  });

  describe('#isPropertySelected', () => {
    const wrapper = shallow(<PropertyContainer properties={propertyList} />)
    wrapper.setState({selectedProperties: [1,2,3]})
    const instance = wrapper.instance()
    it('is true when a property is selected', () => {
      expect(instance.isPropertySelected(2)).toEqual(true)
    })

    it('is false when a property is not selected', () => {
      expect(instance.isPropertySelected(8)).toEqual(false)
    })
  });

  describe('#addProperty', () => {
    it('adds a property to the selectedProperty array', () => {
      const wrapper = shallow(<PropertyContainer properties={propertyList} />)
      wrapper.setState({ selectedProperties: [1, 2, 3] })
      const instance = wrapper.instance()
      instance.addProperty(5)
      expect(instance.state.selectedProperties).toEqual([1,2,3,5])
    })
  });

  describe('#removeProperty', () => {
    it('removes a property to the selectedProperty array', () => {
      const wrapper = shallow(<PropertyContainer properties={propertyList} />)
      wrapper.setState({ selectedProperties: [1, 2, 3] })
      const instance = wrapper.instance()
      instance.removeProperty(3)
      expect(instance.state.selectedProperties).toEqual([1, 2])
    })
  });

});
