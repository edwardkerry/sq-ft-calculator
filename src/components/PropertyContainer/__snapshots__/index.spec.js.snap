// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property Container renders 1`] = `
<ul
  className="properties"
>
  <li
    className="property"
    onClick={[Function]}
  >
    <p
      className="address"
    >
      Address: 
      test address 1
    </p>
    <p
      className="price"
    >
      Value: £
      10000
    </p>
    <p
      className="floor_area"
    >
      Floor area(sq ft): 
      100
    </p>
  </li>
  <li
    className="property"
    onClick={[Function]}
  >
    <p
      className="address"
    >
      Address: 
      test address 2
    </p>
    <p
      className="price"
    >
      Value: £
      20000
    </p>
    <p
      className="floor_area"
    >
      Floor area(sq ft): 
      200
    </p>
  </li>
</ul>
`;
